pkg/operator/baremetal_config.go

```diff
@@ -76,6 +84,31 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		glog.Errorf("provisioningOSDownloadURL not found in Baremetal provisioning CR %s", configName)
 		return BaremetalProvisioningConfig{}, err
 	}
+	// If provisioningNetwork is not provided, set it to a default of "managed".
+	provisioningNetwork, foundNetwork, err := unstructured.NestedString(provisioningSpec, "provisioningNetwork")
+	if !foundNetwork {
+		glog.Infof("provisioningNetwork not found in Baremetal provisioning CR %s", configName)
+		if !provisioningDHCPExternal {
+			provisioningNetwork = provisioningNetworkManaged
+		} else if provisioningDHCPExternal {
+			provisioningNetwork = provisioningNetworkUnmanaged
+		}
+	}
+	if err != nil {
+		glog.Errorf("provisioningNetwork not found in Baremetal provisioning CR %s", configName)
+		return BaremetalProvisioningConfig{}, err
+	}
+	provisioningNetworkCIDR, found, err := unstructured.NestedString(provisioningSpec, "provisioningNetworkCIDR")
+	if !found || err != nil {
+		glog.Errorf("provisioningNetworkCIDR not found in Baremetal provisioning CR %s", configName)
+		return BaremetalProvisioningConfig{}, err
+	}
+
+	// Checking for interdependecies within the config
+	if !foundDHCP && !foundNetwork {
+		glog.Errorf("Both the provisioningNetwork and provisioningDHCPExternal are not found in baremetal provisioning CR %s", configName)
+		return BaremetalProvisioningConfig{}, err
```

Comment:
Will err be set here? I think we would have returned with a non-nil err on line 104 already, right?

Reply:
Yes. We would make our own error here.
Alternatively, we could tolerate a provisioning CR that has both these values missing. We could pick a default (provisioningNetwok = "Managed") and keep going.